src/scripts/plc.ts

```diff
@@ -58,49 +59,57 @@ export async function plcSetupLabeler(options: PlcSetupLabelerOptions) {
 	const publicKey = k256.getPublicKey(privateKey);
 	const keyDid = formatDidKey(SECP256K1_JWT_ALG, publicKey);
 
-	const operation: ComAtprotoIdentitySignPlcOperation.Input = {};
+	const operation: InferXRPCBodyInput<ComAtprotoIdentitySignPlcOperation.mainSchema["input"]> =
+		{};
 
 	const credentials = await agent.get("com.atproto.identity.getRecommendedDidCredentials", {});
 
-	if (
-		!credentials.data.verificationMethods
-		|| !(typeof credentials.data.verificationMethods === "object")
-		|| !("atproto_label" in credentials.data.verificationMethods)
-		|| !credentials.data.verificationMethods["atproto_label"]
-		|| (credentials.data.verificationMethods["atproto_label"] !== keyDid
-			&& options.overwriteExistingKey)
-	) {
-		operation.verificationMethods = {
-			...(credentials.data.verificationMethods || {}),
-			atproto_label: keyDid,
-		};
-	}
-
-	if (
-		!credentials.data.services
-		|| !(typeof credentials.data.services === "object")
-		|| !("atproto_labeler" in credentials.data.services)
-		|| !credentials.data.services["atproto_labeler"]
-		|| typeof credentials.data.services["atproto_labeler"] !== "object"
-		|| !("endpoint" in credentials.data.services["atproto_labeler"])
-		|| credentials.data.services["atproto_labeler"].endpoint !== options.endpoint
-	) {
-		operation.services = {
-			...(credentials.data.services || {}),
-			atproto_labeler: { type: "AtprotoLabeler", endpoint: options.endpoint },
-		};
+	if (credentials.ok) {
```

Comment:
It may make more sense to fail here if `credentials` is not `ok`?

Reply:
I think so, we're just going to exit early on the `Object.keys()` check anyways if not `credentials.ok`